torch/distributed/tensor/placement_types.py

```diff
@@ -109,32 +109,39 @@ def _split_tensor(
         return shard_list, pad_sizes
 
     @staticmethod
-    def _local_shard_size_on_dim(
-        size_on_dim: int,
+    def _local_shard_size_and_offset(
+        curr_local_size: int,
         num_chunks: int,
         rank: int,
-        return_offset: bool = False,
     ) -> tuple[int, int]:
         """
-        returns the local shard size and offset on a given tensor dim
+        Given the size of the current local tensor (which may already be sharded on some dimensions),
+        computes the new local shard size and offset given the desired number of chunks
+        (generally equal to the size of the current sharding dim).
+
+        # TODO(whc) is offset computed relative to the global tensor shape or relative to the original 'local'
```

Comment:
The offset is computed relative to the `curr_local_size`, no global tensor shape is visible here